syntax.cf

```diff
@@ -29,14 +29,15 @@ Multiply. ArithExpr ::= "(" "*" [ArithExpr] ")" ;
 separator nonempty ArithExpr "" ;
 
 -- A comparative operator is a binary operator that compares two inputs
-GreaterThan. BoolExpr ::= "(" ">" ArithExpr ArithExpr ")" ;
-LessThan. BoolExpr ::= "(" "<" ArithExpr ArithExpr ")" ;
-GreaterEqual. BoolExpr ::= "(" ">=" ArithExpr ArithExpr ")" ;
-LessEqual. BoolExpr ::= "(" "<=" ArithExpr ArithExpr ")" ;
-NotEqual. BoolExpr ::= "(" "!=" ArithExpr ArithExpr ")" ;
-Equal. BoolExpr ::= "(" "==" ArithExpr ArithExpr ")";
+GreaterThan. CompExpr ::= "(" ">" ArithExpr ArithExpr ")" ;
+LessThan. CompExpr ::= "(" "<" ArithExpr ArithExpr ")" ;
+GreaterEqual. CompExpr ::= "(" ">=" ArithExpr ArithExpr ")" ;
+LessEqual. CompExpr ::= "(" "<=" ArithExpr ArithExpr ")" ;
+NotEqual. CompExpr ::= "(" "!=" ArithExpr ArithExpr ")" ;
+Equal. CompExpr ::= "(" "==" ArithExpr ArithExpr ")";
```

Comment:
Apart from keeping `CompExpr`, folded into `BoolExpr` as we discussed, this looks good to me!